include/tvm/relay/expr.h

```diff
@@ -166,12 +166,20 @@ class VarNode : public ExprNode {
 
 RELAY_DEFINE_NODE_REF(Var, VarNode, Expr);
 
+/*! \brief Hash Var by it's id.
+ * Different VarNode might has same vid, and they are considered to be the same var in such case.
+ * Use VarHash to hash Var by id.
+ */
 struct VarHash {
```

Comment:
On second thought, while I know that Var can be defined by using identity in id, it is dangerous to make this assumption. Most of the current code will keep a unique Var during transformation(and transform all occurrences of Var if its type signature change) So likely we don't need to rely on the comparison by id within a pass. Please check if the pass can be improved.

Reply:
No It cannot. Typesubst get called multiple time. I can write a pass that bind them back to a single one, but I found out I dont need to for our test case right now, as dce is working fine.